chainbench/test_data/solana.py

```diff
@@ -0,0 +1,101 @@
+from tenacity import retry, stop_after_attempt
+
+from chainbench.test_data.base import BaseTestData, Block, BlockchainData
+from chainbench.util.rng import get_rng
+
+
+class SolanaTestData(BaseTestData):
+    TXS_REQUIRED = 100
+    ACCOUNTS_REQUIRED = 200
+    BLOCK_TIME = 0.4
+
+    def _fetch_block(
+        self, block_number: int, return_txs: bool = True
+    ) -> dict | Exception:
+        if return_txs:
+            transaction_details = "accounts"
+        else:
+            transaction_details = "none"
+        config_object = {
+            "encoding": "json",
+            "transactionDetails": transaction_details,
+            "rewards": False,
+            "maxSupportedTransactionVersion": 0,
+        }
+        try:
+            result = self._make_call("getBlock", [block_number, config_object])
+        except Exception as e:
+            self._logger.error(f"Failed to fetch block {block_number}: {e}")
+            return e
+        return result
+
+    @retry(reraise=True, stop=stop_after_attempt(5))
+    def _fetch_random_block(self, start, end, return_txs=True) -> dict:
+        rng = get_rng()
+        block_number = rng.random.randint(start, end)
+        block = self._fetch_block(block_number, return_txs=return_txs)
+        if isinstance(block, Exception):
+            raise block
+        else:
+            return block
+
+    def _fetch_latest_slot_number(self):
+        slot = self._make_call("getLatestBlockhash")["context"]["slot"]
+        return slot
+
+    def _fetch_latest_block(self):
+        slot_number = self._fetch_latest_slot_number()
+        latest_block = self._fetch_block(slot_number, return_txs=True)
+        return slot_number, latest_block
+
+    def _fetch_first_available_block(self):
+        block = self._make_call("getFirstAvailableBlock")
+        return block
+
+    # get initial data from blockchain
+    def _get_init_data(self, parsed_options) -> BlockchainData:
+        txs: list[dict] = []
+        tx_hashes: list[str] = []
+        accounts: set[str] = set()
+        blocks: list[Block] = []
+        end_block_number, _latest_block = self._fetch_latest_block()
+        start_block_number = self._fetch_first_available_block()
+
+        # factor in run_time and add 10% buffer to ensure blocks used in test data are
+        # not removed from the ledger
+        start_block_number += int((parsed_options.run_time / self.BLOCK_TIME) * 1.1)
+
+        while self.TXS_REQUIRED > len(txs) or self.ACCOUNTS_REQUIRED > len(accounts):
+            if self.ACCOUNTS_REQUIRED > len(accounts) or self.TXS_REQUIRED > len(
+                blocks
+            ):
```

Comment:
splitting `len(blocks)` to 3 lines looks terrible. I understand that it's linter, maybe help him with smth like 
```python
if (
  self.ACCOUNTS_REQUIRED > len(accounts) 
  or self.TXS_REQUIRED > len(blocks)
):

Reply:
Ammended to:
```
            if (
                    self.ACCOUNTS_REQUIRED > len(accounts)
                    or self.TXS_REQUIRED > len(blocks)
            ):
```